PROTO_tests/tests/pdarray_creation_test.py

```diff
@@ -0,0 +1,742 @@
+        assert [0.30013431967121934, 0.47383036230759112, 1.0441791878997098] == u_array.to_list()
+
+        u_array = ak.uniform(size=np.int64(3), low=np.int64(0), high=np.int64(5), seed=np.int64(0))
+        assert [0.30013431967121934, 0.47383036230759112, 1.0441791878997098] == u_array.to_list()
+
+        with pytest.raises(TypeError):
+            ak.uniform(low="0", high=5, size=size)
+
+        with pytest.raises(TypeError):
+            ak.uniform(low=0, high="5", size=size)
+
+        with pytest.raises(TypeError):
+            ak.uniform(low=0, high=5, size="100")
+
+        # Test that int_scalars covers uint8, uint16, uint32
+        uint_arr = ak.uniform(low=np.uint8(0), high=np.uint16(5), size=np.uint32(100), seed=np.uint8(1))
+        int_arr = ak.uniform(low=0, high=5, size=100, seed=1)
+        assert (uint_arr == int_arr).all()
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", [ak.int64, float, ak.float64, bool, ak.bool, ak.bigint])
+    def test_zeros_dtype(self, size, dtype):
+        zeros = ak.zeros(size, dtype)
+        assert isinstance(zeros, ak.pdarray)
+        assert dtype == zeros.dtype
+        assert (0 == zeros).all()
+
+    def test_zeros_misc(self):
+        zeros = ak.ones("5")
+        assert 5 == len(zeros)
+
+        with pytest.raises(TypeError):
+            ak.zeros(5, dtype=ak.uint8)
+
+        with pytest.raises(TypeError):
+            ak.zeros(5, dtype=str)
+
+        # Test that int_scalars covers uint8, uint16, uint32, str
+        int_arr = ak.zeros(5)
+        for arg in np.uint8(5), np.uint16(5), np.uint32(5), str(5):
+            assert (int_arr == ak.zeros(arg, dtype=ak.int64)).all()
+
+    @pytest.mark.parametrize("dtype", [int, ak.int64, float, ak.float64, bool, ak.bool, ak.bigint])
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    def test_ones_dtype(self, size, dtype):
+        ones = ak.ones(size, dtype)
+        assert isinstance(ones, ak.pdarray)
+        assert dtype == ones.dtype
+        assert (1 == ones).all()
+
+    def test_ones_misc(self):
+        ones = ak.ones("5")
+        assert 5 == len(ones)
+
+        with pytest.raises(TypeError):
+            ak.ones(5, dtype=ak.uint8)
+
+        with pytest.raises(TypeError):
+            ak.ones(5, dtype=str)
+
+        # Test that int_scalars covers uint8, uint16, uint32
+        int_arr = ak.ones(5)
+        for arg in np.uint8(5), np.uint16(5), np.uint32(5):
+            assert (int_arr == ak.ones(arg, dtype=ak.int64)).all()
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", [ak.int64, ak.float64, ak.bool, ak.bigint])
+    def test_ones_like(self, size, dtype):
+        int_ones = ak.ones(size, dtype)
+        int_ones_like = ak.ones_like(int_ones)
+
+        assert isinstance(int_ones_like, ak.pdarray)
+        assert dtype == int_ones_like.dtype
+        assert (1 == int_ones_like).all()
```

Comment:
this is super nitpicky, but since we're parameterizing on all dtypes, maybe we should choose another name than `int_ones_like`